### ISSUE TITLE
having error  about  code in icarl

### ISSUE THREAD

**Issue by dota2mhxy** [G-U-N/PyCIL#24]
Thank you for your work on class incremental code integration. It's wonderful. I found a problem in the icarl code.
 Loss is composed of the classification of new classes and the distillation of knowledge of old classes. But your code wrongly changed the classification of new classes to the classification of all classes。

![image](https://user-images.githubusercontent.com/64951554/210311655-8b45d68a-34dd-4c4a-9a15-d55b393317c9.png)
![image](https://user-images.githubusercontent.com/64951554/210311684-606599ff-1cc8-4b29-a25c-f82a728ec642.png)

**Comment by zhoudw-zdw** [comment]
It seems your question is about the output dimension adopted in the cross-entropy loss.

You can modify the dimension like [this](https://github.com/G-U-N/PyCIL/blob/master/models/finetune.py#L151) in the loss calculation and report your findings, if applicable.

**Comment by kleinzcy** [comment]
Thank your nice codebase. I am confused about the loss function of iCaRL. In the origin paper, they adopt sigmoid function to calculate loss instead of softmax function. In my understanding, the two forms is not the same. So, the question is why you adopt cross-entropy loss in iCaRL?

**Comment by zhoudw-zdw** [comment]
> Thank your nice codebase. I am confused about the loss function of iCaRL. In the origin paper, they adopt sigmoid function to calculate loss instead of softmax function. In my understanding, the two forms is not the same. So, the question is why you adopt cross-entropy loss in iCaRL?

You are correct. Just change the CE loss into BCE loss for a try. In my observation the performance would improve when using CE loss.
